### PR TITLE
Rewrite queue to make it more meaningful and readable, add type hints…

### PR DESCRIPTION
**Pull request by king-phyte**
### Describe your change:

* Replaced previous definition of the Queue class with a more readable one
* Added more meaningful functions
* Added type hints
* Added doctests to the class

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: data_structures/queue/queue_on_list.py [hunk 1]
```diff
@@ -1,52 +1,77 @@
...
+    def size(self) -> int:
+        return len(self._queue)
+
+    def is_empty(self) -> bool:
+        return self.size == 0
+
+    def rotate(self, number_of_rotations: int) -> None:
+        """Rotate a queue by `number_of_rotation` times"""
+        for _ in range(number_of_rotations):
+            self.enqueue(self.dequeue())
```
Comment (IdoErel) [inline data_structures/queue/queue_on_list.py:77]: Can be executed more efficiently -
by combining it into a one-liner,
or using `repeat` from `itertools`.

**Review by IdoErel — APPROVED** [review]
Awesome documentation, better code. Much cleaner now ;D


Kind regards, 
IdoErel

File: data_structures/queue/queue_on_list.py (hunk 1 above)
Comment (king-phyte) [reply data_structures/queue/queue_on_list.py:77]: You are right. I didn't go with the one-liner for readability sake, but I love your idea about itertools.

Thank you.

**Comment by king-phyte** [comment]
Please I see the PR hasn't been merged. Are there any reasons why? Anything you'd like me to add or modify?

**Comment by enedil** [comment]
@IdoErel mind approving?

**Comment by enedil** [comment]
Screw you, Mr stale bot

**Comment by enedil** [comment]
Screw you, Mr stale bot

**Pull request closed without merging**